pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -102,46 +87,12 @@ func (c TargetConfigController) sync(ctx context.Context, syncContext factory.Sy
 		return nil
 	}
 
-	infra, err := c.infraLister.Get(infraGlobalName)
-	if err != nil {
-		return err
-	}
+	globalSharedState := utils.GetGlobalSharedState()
+	configMapState := globalSharedState.GetCSIConfigState()
 
-	clusterCSIDriver, err := c.clusterCSIDriverLister.Get(utils.VSphereDriverName)
-	if err != nil {
-		return err
-	}
-
-	// TODO: none of our CSI operators check whether they are running in the correct cloud. Is
-	// this something we want to change? These operators are supposed to be deployed by CSO, which
-	// already does this checking for us.
-
-	cloudConfig := infra.Spec.CloudConfig
-	cloudConfigMap, err := c.configMapLister.ConfigMaps(cloudConfigNamespace).Get(cloudConfig.Name)
-	if err != nil {
-		return fmt.Errorf("failed to get cloud config: %w", err)
-	}
-
-	cfgString, ok := cloudConfigMap.Data[infra.Spec.CloudConfig.Key]
-	if !ok {
-		return fmt.Errorf("cloud config %s/%s does not contain key %q", cloudConfigNamespace, cloudConfig.Name, cloudConfig.Key)
-	}
-
-	var cfg vsphere.VSphereConfig
-	err = gcfg.ReadStringInto(&cfg, cfgString)
-	if err != nil {
-		return err
-	}
-
-	requiredCM, err := c.applyClusterCSIDriverChange(infra, cfg, clusterCSIDriver)
-	if err != nil {
-		return err
-	}
-
-	// TODO: check if configMap has been deployed and set appropriate conditions
-	_, _, err = resourceapply.ApplyConfigMap(ctx, c.kubeClient.CoreV1(), syncContext.Recorder(), requiredCM)
-	if err != nil {
-		return err
+	// if configmap is not created simply return
+	if !configMapState {
```

Comment:
@jsafrane @bertinatto please check this. This will cause targetconfig operator to only set conditions once configmaps is created. I have been debating, if we should remove this controller entirely. 

Could we miss a case where cluster was upgraded but has left over conditions?

Reply:
There is [RemoveStaleConditionsController](https://github.com/openshift/library-go/blob/a467dafcef6988afbcce43400c339d7ca68a3298/pkg/operator/staleconditions/remove_stale_conditions.go#L19) that can remove orphaned conditions

Reply:
Added a variant of it, so as stale conditions can be removed.

Reply:
This controller won't do anything until a config map is loaded, it will not even update the old conditions. How useful are these conditions then? Can we remove them right away?

Reply:
I have updated the code to remove older conditions right away.

Reply:
I should have made myself more obvious, can you please remove the whole `TargetConfigController` and just run [RemoveStaleConditionsController](https://github.com/openshift/library-go/blob/a467dafcef6988afbcce43400c339d7ca68a3298/pkg/operator/staleconditions/remove_stale_conditions.go#L19) instead? Less code, more re-use.

Reply:
The code in both the controllers are not the same. I do not think, it makes sense to remove conditions every 1 minute.